Source/WebCore/dom/MouseEvent.idl

```diff
@@ -44,13 +44,12 @@
     readonly attribute double movementY;
     [ImplementedAs=force] readonly attribute double webkitForce;
 
-    // FIXME: relatedTarget should be of type EventTarget? but we need to use any to support a quirk for iAd Producer (rdar://problem/30640101).
-    [CallWith=CurrentGlobalObject&CurrentScriptExecutionContext, ImplementedAs=initMouseEventQuirk] undefined initMouseEvent([AtomString] DOMString type,
-        optional boolean canBubble = false, optional boolean cancelable = false,
+    // FIXME: Using "undefined" as default parameter value is wrong.
+    undefined initMouseEvent(optional [AtomString] DOMString type = "undefined", optional boolean canBubble = false, optional boolean cancelable = false,
```

Comment:
I don't really understand why `type` is changed from a required argument to an optional argument defaulting to "`undefined`". This is also not explained by the commit message.

Reply:
Web-Spec - https://w3c.github.io/uievents/#idl-interface-MouseEvent-initializers ; we should just sync this.

Reply:
Notwithstanding aligning with the spec is good, it probably makes more senes to remove the quirk separate to making other WebIDL changes.

Reply:
yes it's what I was trying to do initially, without success. 
Just "reverting" is not working. 
Note that this is the initial code before the Quirk…
https://github.com/WebKit/WebKit/commit/51b46788f5aa0519a2c829cd6a01569e468ff568

Reply:
But at some later point that was tightened up, no? Because you are changing this away from `[AtomString] DOMString type` after all.

What exactly is not working?